src/main/resources/pl/edu/agh/iet/dts/ui/static/js/controller/mapController.js

```diff
@@ -1,11 +1,108 @@
+            function (response) {
+                $scope.assignDate();
+                $scope.createRoute(response.data.data);
+            },
+            function (error) {
+                console.log(error);
+            }
+        );
     };
 
-    $scope.showPreferences = function () {
-        $location.path('/app/components/user-form').replace();
+    $scope.assignDate = function () {
+        const format = function (num) {
+            return ('0' + num).slice(-2);
+        };
+
+        now = new Date();        
+        date = now.getFullYear() + '-' + format(now.getMonth() + 1) + '-' + format(now.getDate());
+        time = format(now.getHours()) + ':' + format(now.getMinutes());
+        $rootScope.lastSync = date + ' ' + time;
     };
 
-});
+    $scope.createRoute = function (data) {
+        if (checkpointArray.length === data.length && checkpointLimit === $rootScope.points) {
+            return;
+        }
+
+        checkpointLimit = $rootScope.points;
+        checkpointArray = data.slice();
+        let routeRequest;
+
+        num = Math.min($rootScope.points, 10)
+        data = data.slice(Math.max(0, data.length - num));
+
+        if (data.length === 0) {
+            console.log('Nothing to show. There is no GPS data for this user.');
+        } else if (data.length === 1) {
+            let singlePointCoordinates = new google.maps.LatLng(data[0].latitude, data[0].longitude);
+
+            routeRequest = {
+                origin: singlePointCoordinates,
+                destination: singlePointCoordinates,
+                waypoints: [],
+                travelMode: travelMode
+            };
+        } else {
+            let startPoint = data.shift();
+            let endPoint = data.pop();
+            let wayPoints = [];
+
+            data.forEach(function (wayPoint) {
+                wayPoints.push({
+                    location: new google.maps.LatLng(wayPoint.latitude, wayPoint.longitude),
+                    stopover: wayPoint.stop
+                });
+            });
+
+            routeRequest = {
+                origin: new google.maps.LatLng(startPoint.latitude, startPoint.longitude),
+                destination: new google.maps.LatLng(endPoint.latitude, endPoint.longitude),
+                waypoints: wayPoints,
+                travelMode: travelMode
+            };
+        }
+
+        $scope.displayRoute(routeRequest);
+    };
+
+    $scope.displayRoute = function (routeRequest) {
+        directionsService.route(routeRequest, function (response, status) {
+            if (status === google.maps.DirectionsStatus.OK) {
+                directionsDisplay.setDirections(response);
+            } else {
+                console.log('Response to Google Maps API failed due to: ' + status + '.');
+            }
+        });
+    };
+
+    $rootScope.locationRequest();
+    $rootScope.promise = $interval($rootScope.locationRequest, $rootScope.aggregationTime * 1000);
```

Comment:
The `aggregationTime` is given in minutes, please modify the scheduler's interval.